post.md

```diff
@@ -0,0 +1,346 @@
+0x100. However, nowadays we use more sophisticated formats
+because modern CPUs offer access rights for memory regions
+so that we can disable write access for code and constants
+and execution for data regions. Another reason is that often
+we want to load shared libraries — the code that can be
+shared between processes. Of course, it's possible to load
+them using dedicated system calls, but it's much more
+convenient to let executable loader do it for us. Except
+that, we want to attach debugging symbols for convenient
+execution analysis. Moreover, modern executable formats
+contain information about target architecture, checksums
+and other information about the code.
+
+Except that, producing binaries directly from source code
+would be very impractical and inefficient for big projects.
+That's why in modern systems we have at least 4 file types
+for code:
+
+   1. Object files — usually generated per module,
+   containing functions with their data. At this point, no 
+   function dependencies are checked so that we may take
+   care of later. Thanks to this we can separate compilation 
+   process from resolving dependencies. Extension *.o in
+   *nix and *.obj in Windows.
+   2. Static libraries — set of functions to be incorporated
+   into executable. Unlike object files, they must contain
+   all dependencies. Extension *.a in *nix and *.lib in
+   Windows.
+   3. Dynamic libraries — a special type of library which is
+   suitable to be loaded once for many programs at the same
+   time. In such case, no their code is incorporated into
+   binary but only references to them. Such code doesn't have
+   to be duplicated in RAM too. However, each process has
+   separate space for data. Unlike static libraries, they 
+   have also initialization code that is run when the library is
+   loaded. They may be loaded at the same time as whole
+   binary or during runtime using the system call. The second
+   option is often used to deploy plugins. However this
+   approach is convenient, it may cause problems with
+   dependencies because of many versions of the same library.
+   That's why many (especially closed source) are
+   distributed with libraries incorporated into the binary
+   (ie. statically linked). Extension *.so in Linux, *.dll
+   in Windows,  *.dylib in *BSD (including Mac). In *nix
+   systems they are usually stored in /lib /usr/lib (this
+   can be reconfigured per system or per binary). Windows
+   usually store them in Windows and Windows\System32
+   directory, but the directory with binary is checked by
+   default.
+   4. Executables — binaries intended to be run as
+   standalone programs. They usually accept command line
+   parameters and environment variables as an input. In
+   *nix system they usually have no extension, in Windows
+   *.exe.
+
+![shared_vs_static](shared.jpg)
+graphic from [here](https://medium.com/@romalms10/why-dynamic-libraries-bbaa55b199db)
```

Comment:
This link does not seem to work. You did not provide a full image address. For now it is only shared.jpg. I suppose it should be ![image](https://cdn-images-1.medium.com/max/800/1*haBSlrgPuetSGVaDU-TrFA.jpeg "shared_vs_static") this one

Reply:
now it should work

Reply:
Uploading a jpg file is unnecessary. What I meant before was to correct the link address to `[shared_vs_static](https://cdn-images-1.medium.com/max/800/1*haBSlrgPuetSGVaDU-TrFA.jpeg "shared_vs_static") ` since it is the correct graphics.

Also, please rename the post.md (do not add post.md in the general catalogue) and put it in _posts folder. For example, the post can be named: `2018-04-19-the-art-of-disassembly.md`